shared/js/search_provider.js

```diff
@@ -32,6 +32,7 @@
 
   // Allow consumers to wait for data to be initialised
   var readyPromise;
+  var resolver;
```

Comment:
This and ```readyPromise``` should probably be a member of the class instead of a single reference in the closure. It's unlikely to cause problems though, so we can fix this later.